soh/soh/OTRGlobals.cpp

```diff
@@ -1526,14 +1665,62 @@ extern "C" int Randomizer_CopyHintFromCheck(RandomizerCheck check, char* buffer,
     return CopyStringToCharBuffer(hintText, buffer, maxBufferSize);
 }
 
+extern "C" GetItemEntry ItemTable_Retrieve(int16_t getItemID) {
+    if (OTRGlobals::Instance->getItemModIndex != MOD_VANILLA) {
+        getItemID++; // counteracts the - 1 offset used for vanilla table
+    }
+
+    return ItemTableManager::Instance->RetrieveItemEntry(OTRGlobals::Instance->getItemModIndex, getItemID);
+}
+
+extern "C" GetItemEntry ItemTable_RetrieveEntry(s16 tableID, s16 getItemID) {
+    if (tableID == MOD_VANILLA) {
+        getItemID--; // counteracts the - 1 offset used for vanilla table
+    }
+    return ItemTableManager::Instance->RetrieveItemEntry(tableID, getItemID);
+}
+
 extern "C" s32 Randomizer_GetRandomizedItemId(GetItemID ogId, s16 actorId, s16 actorParams, s16 sceneNum) {
+    if (OTRGlobals::Instance->gRandomizer->CheckContainsRandoItem(OTRGlobals::Instance->gRandomizer->GetCheckFromActor(sceneNum, actorId, actorParams))) {
+        OTRGlobals::Instance->getItemModIndex = MOD_RANDOMIZER;
+    } else {
+        OTRGlobals::Instance->getItemModIndex = MOD_VANILLA;
+    }
```

Comment:
Right now there's a mix of this way and the fixed way, since this way works like 95% of the time and changing everything would take a while. Meanwhile people can use this for keysanity and such.